subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/execution/ProgramParser.kt

```diff
@@ -36,7 +36,7 @@ object ProgramParser {
         val topLevelBlockIds =
             when (target) {
                 ProgramTarget.Project -> arrayOf("buildscript", "plugins")
-                ProgramTarget.Settings -> arrayOf("buildscript", "pluginManagement")
+                ProgramTarget.Settings -> arrayOf("buildscript", "pluginManagement", "plugins")
                 ProgramTarget.Gradle -> arrayOf("initscript")
```

Comment:
❓ I think this should be moved onto a field in `ProgramTarget`.

Reply:
Nope. In general, keep the decision making code in a single place so the control flow is apparent.

Reply:
The problem with doing that though is that it breaks the Single Responsibility Principle. As a contributor coming in to change this logic, I had to figure out the three different locations I needed to change to add support for a new top level scripting block type.

Instead, being able to have one common location with all of this knowledge would mean one simple place to change in order to (for example) add support for `plugins` on `init` scripts.

Thoughts?

Reply:
I see a single place where the `"plugins"` string had to be added.

Reply:
This has been resolved a bit cleaner without touching `ProgramTarget`